### PR TITLE
Refactored `PaymentButton` name - `Edges` to `Shape`

### PR DESCRIPTION
**Pull request by stechiu**
### Reason for changes

JS and Android don't use the parameter `edges` to describe their button shapes. Instead, they use the parameter `shape` and the values `rectangle` (`rect` for JS), `rounded`, and `pill`. This PR is for parity in naming among iOS, Android, and JS.

### Summary of changes

| Before | After |
| ----------- | ----------- |
| `PaymentButtonEdges` | `PaymentButtonShape` |
| `hardEdges` | `rectangle` |
| `softEdges` | `rounded` |
| `rounded` | `pill` | 

### Checklist

- [x] Added a changelog entry

- @stechiu

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +19,9 @@
   * Add `PayPalVaultResult` type to return vault result
   * Add `PayPalVaultDelegate` to handle results from vault flow
   * Add `PayPalWebCheckoutClientError.paypalVaultResponseError` for missing or invalid response from vaulting
+* Breaking Changes
```
Comment (jaxdesmarais) [inline CHANGELOG.md:21]: Should this be targeting `next-major-version` instead of the `main` branch?

**Review by richherrera — APPROVED** [review]

**Review by agedd — APPROVED** [review]

**Review by jaxdesmarais — CHANGES_REQUESTED** [review]
Blocking as this is currently a breaking change on main

File: CHANGELOG.md (hunk 1 above)
Comment (stechiu) [reply CHANGELOG.md:22]: It can be. Will the next update be a major or minor version update? Currently Android and JS have the updated shape names

Comment (stechiu) [reply CHANGELOG.md:21]: From the looks of the current CHANGELOG, all breaking changes go in major version releases. I've made the updates accordingly in CHANGELOG and the PR merges into `next-major-version`

Comment (jaxdesmarais) [reply CHANGELOG.md:21]: Thanks for making that update! Ywp, what is merged into `main` as unreleased will be released in a minor version. What is merged into `next-major-version` will contain breaking changes and will be released in a major version.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,23 +19,10 @@
...
-  `.topUpWith`
-  * `PayPalButtonSize` `.mini` changed to rectangular button to meet brand guidelines
-  * Font typeface changed to "PayPalOpen" to meet brand guidelines
 * Breaking Changes
-  * PaymentButtons
-    * Remove `PayPalButton.Labels` cases: `.checkout`, `.buyNow`, `.payLater`
-    * Remove `PaymentButtonColor` `.black`, `.silver`, `.blue`, and `.darkBlue`
-    * Add `PaymentButtonColor.gold` for `PayPalCreditButton`
-    * Remove `PaymentButtonSize` cases: `.miniWithWordmark`, `.collapsed`, `.expanded`, `.full`
+  * PaymentButton
```
Comment (jaxdesmarais) [inline CHANGELOG.md]:
```suggestion
  * PaymentButtons
```

File: Demo/Demo/Models/Order.swift [hunk 3]
```diff
@@ -36,9 +36,9 @@ struct Order: Codable, Equatable {
 
     struct Vault: Codable, Equatable {
         
-        let id: String
+        let id: String?
```
Comment (jaxdesmarais) [inline Demo/Demo/Models/Order.swift]: Should this change be in this PR? It looks to be unrelated.

File: Demo/Demo/SwiftUIComponents/CardPaymentViews/CardOrderCompletionResultView.swift [hunk 4]
```diff
@@ -43,11 +43,15 @@ struct CardOrderCompletionResultView: View {
                 LeadingText("Brand", weight: .bold)
                 LeadingText("\(brand)")
             }
+            if let vaultStatus = orderResponse.paymentSource?.card?.attributes?.vault.status {
```
Comment (jaxdesmarais) [inline Demo/Demo/SwiftUIComponents/CardPaymentViews/CardOrderCompletionResultView.swift]: Same question here, not sure if these diffs should be in this PR

File: Sources/PaymentButtons/PayPalButton.swift [hunk 5]
```diff
@@ -8,6 +8,9 @@ public final class PayPalButton: PaymentButton {
     public enum Color: String {
         case gold
         case white
+        case black
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PayPalButton.swift]: Looks like these changes may be from another PR too

File: Demo/Demo/Models/Order.swift (hunk 3 above)
Comment (stechiu) [reply Demo/Demo/Models/Order.swift]: When I switched from pointing to `next-major-version` from `main`, there were several conflicts that I resolved. I checked, and this change, along with others that you commented on, already exist on `next-major-version`. I didn't pull in the latest from `main` into `next-major-version` branch before creating this new branch `fix/button-edges-name`. Are you ok with me merging these changes into `next-major-version`?

File: CHANGELOG.md (hunk 1 above)
Comment (jaxdesmarais) [reply CHANGELOG.md:21]: Have you had a chance to push those changes up yet to `next-major-version`? It looks like last push to [that branch](https://github.com/paypal/paypal-ios/tree/next-major-version) was ~2 weeks ago. Happy to pair on this this afternoon or tomorrow if you'd like!

**Comment by stechiu** [comment]
> Blocking as this is currently a breaking change on main

@jaxdesmarais Issues should be fixed now. I had originally branched from `main` and when I changed the branch to point to `next-major-version`, there were many bad merges. Please check to make sure I got them all!

File: Sources/PaymentButtons/PayPalButton.swift [hunk 6]
```diff
@@ -37,20 +37,20 @@ public final class PayPalButton: PaymentButton {
     /// - Parameters:
     ///   - insets: Edge insets of the button, defining the spacing of the button's edges relative to its content.
     ///   - color: Color of the button. Default to gold if not provided.
-    ///   - edges: Edges of the button. Default to softEdges if not provided.
-    ///   - size: Size of the button. Default to standard if not provided.
+    ///   - shape: Shape of the button. Default to `.rounded` if not provided.
+    ///   - size: Size of the button. Default to collapsed if not provided.
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PayPalButton.swift]:
```suggestion
    ///   - size: Size of the button. Default to `.standard` if not provided.
```

File: Sources/PaymentButtons/PayPalButton.swift [hunk 7]
```diff
@@ -73,21 +73,21 @@ public extension PayPalButton {
         /// - Parameters:
         ///   - insets: Edge insets of the button, defining the spacing of the button's edges relative to its content.
         ///   - color: Color of the button. Default to gold if not provided.
-        ///   - edges: Edges of the button. Default to softEdges if not provided.
-        ///   - size: Size of the button. Default to standard if not provided.
+        ///   - shape: Shape of the button. Default to `.rounded` if not provided.
+        ///   - size: Size of the button. Default to collapsed if not provided.
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PayPalButton.swift]:
```suggestion
        ///   - size: Size of the button. Default to `.standard` if not provided.
```

File: Sources/PaymentButtons/PayPalPayLaterButton.swift [hunk 8]
```diff
@@ -50,19 +50,19 @@ public extension PayPalPayLaterButton {
         /// - Parameters:
         ///   - insets: Edge insets of the button, defining the spacing of the button's edges relative to its content.
         ///   - color: Color of the button. Default to gold if not provided.
-        ///   - edges: Edges of the button. Default to softEdges if not provided.
-        ///   - size: Size of the button. Default to standard if not provided.
+        ///   - shape: Shape of the button. Default to `.rounded` if not provided.
+        ///   - size: Size of the button. Default to collapsed if not provided.
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PayPalPayLaterButton.swift]:
```suggestion
        ///   - size: Size of the button. Default to `.standard` if not provided.
```

File: Sources/PaymentButtons/PaymentButtonShape.swift [hunk 9]
```diff
@@ -32,15 +32,15 @@ public enum PaymentButtonEdges: Equatable {
     }
     public var description: String {
         switch self {
-        case .hardEdges:
+        case .rectangle:
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PaymentButtonShape.swift:35]: Do we want to return "rectangle" here?

**Review by jaxdesmarais — COMMENTED** [review]
Just a few small comments on docstrings. Nice work sorting out the merge issue, I'm not seeing anything you missed from the merge conflicts.

File: Sources/PaymentButtons/PayPalButton.swift (hunk 6 above)
Comment (stechiu) [reply Sources/PaymentButtons/PayPalButton.swift]: Ah missed these, thanks!

File: Sources/PaymentButtons/PaymentButtonShape.swift (hunk 9 above)
Comment (stechiu) [reply Sources/PaymentButtons/PaymentButtonShape.swift:35]: Looks like a typo where the return should be "rectangle" to match the case name. Let me know if you were thinking something different for it

**Review by jaxdesmarais — APPROVED** [review]
👏 thanks for addressing my questions! :shipit:
